pkg/domain/infosync/info.go

```diff
@@ -724,12 +724,21 @@ func (is *InfoSyncer) GetMinStartTS() uint64 {
 	return is.minStartTS
 }
 
+func (is *InfoSyncer) getMinStartTsEtcdCli() *clientv3.Client {
+	if config.GetGlobalConfig().EnableKeyspaceLevelGC {
+		// If `EnableKeyspaceLevelGC` is true,we should use etcd client with keyspace prefix to access min start timestamp.
+		return is.etcdCli
+	}
+	return is.unprefixedEtcdCli
+}
+
 // storeMinStartTS stores self server min start timestamp to etcd.
 func (is *InfoSyncer) storeMinStartTS(ctx context.Context) error {
-	if is.unprefixedEtcdCli == nil {
+	minStartTsEtcdCli := is.getMinStartTsEtcdCli()
+	if minStartTsEtcdCli == nil {
 		return nil
 	}
-	return util.PutKVToEtcd(ctx, is.unprefixedEtcdCli, keyOpDefaultRetryCnt, is.minStartTSPath,
+	return util.PutKVToEtcd(ctx, minStartTsEtcdCli, keyOpDefaultRetryCnt, is.minStartTSPath,
```

Comment:
I have a question that I'm not sure if you've already considered: should unfinished transactions block GC of the null-keyspace and keyspaces that didn't enable keyspace-level-gc?

Reply:
I think in this PR for TiDB, We only accept gc_management_type =keyspace-level-gc when creating a new keyspace, the related judgment logic is in func `CheckTiDBGCManagementTypeAndKeyspaceMeta `, 
and then we can start TiDB for this new keyspace. So all transactions in this keyspace are started with the "keyspace-level-gc".

If someone needs to upgrade the gc management type of any keyspace from "global-gc to "keyspace-level-gc", more operations are required, but the operation are not in this PR.

Reply:
This PR for PD prevents the keyspace from being updated after it has been created,
You can set `gc_management_type` = `keyspace-level-gc` only when you create a keyspace:
https://github.com/tikv/pd/issues/8061

Reply:
Sorry but I guess you misunderstood my question. What I was asking is whether the global GC should be blocked by an unfinished transaction that belong to a specific keyspace.

Reply:
> Sorry but I guess you misunderstood my question. What I was asking is whether the global GC should be blocked by an unfinished transaction that belong to a specific keyspace.

If a specific keyspace is already set "gc_management_type" = "keyspace-level-gc",
this keyspace will not have any impact on the calculation of the global GC.

So the implementation here is that with keyspace level gc enabled, the etcd path of keyspace's minstartts is prefixed with keyspace.